spec/features/18f_procurement_spec.rb

```diff
@@ -27,7 +27,7 @@
       expect(Dispatcher).to receive(:deliver_new_proposal_emails)
 
       visit '/gsa18f/procurements/new'
-      fill_in "Product Name and Description", with: "buying stuff"
+      fill_in "Product name and description", with: "buying stuff"
```

Comment:
Yeah I think the current state is better than using `name` or another field that is not visible to the user.

I would do it by using I18n! so something like

 `fill_in t('forms.product.name_and_description'), with: buying stuff"`

...but if we aren't using a lot of I18n, that might not make sense for this PR